cEP-0031.md

```diff
@@ -0,0 +1,254 @@
+    be using a plugin to solve this issue.
+
+-   [coala/coala-bears#1897][1897]: Show PEP8 error description
+    On running PEP8Bear the default message is 'The code does not comply to
+    PEP8.' which is not very helpful for the user. This issue will be solved
+    by implementing upstream changes to autopep8.
+
+-   [hhatto/autopep8#227][227]: Create diff for a specific pep8 issue
+    User should have a choice of selecting the issue he wants to generate a fix
+    for. Although the issue is a part of the autopep8, we will be solving
+    this as a part of upstream changes to autopep8.
+
+Along with fixing these issues, new bears including `OutdatedDependencyBear`,
+`FileExistsBear`, `RegexLintBear`, `FileModeBear` and `RequirementsCheckBear`
+will be implemented.
+
+## Implementation of new bears
+
+### 1. OutdatedDependencyBear
+
+Issue: [coala/coala-bears#2445][2445]
+
+New functions to the existing package manager would be the best way of
+implementing this.
+
+- For pip requirements, we could use the PyPI JSON API which provides the
+    version info by specifying the `package_name`.
+
+```py
+class PipRequirement(PackageRequirement):
+    def get_version_info(self):
+        # This would get the package info in json format
+        url = 'https://pypi.python.org/pypi/%s/json' % (self.package,)
+
+        # Get the required data
+        data = json.load()
+        versions = data['releases'].keys().sort(key = StrictVersion)
+        return versions
+```
+
+- For npm dependencies, we can use the npm cli command `npm outdated` which
+    produces an output as below:
+
+```
+$ npm outdated
+Package      Current   Wanted   Latest  Location
+glob          5.0.15   5.0.15    6.0.1  test-outdated-output
+nothingness    0.0.3      git      git  test-outdated-output
+npm            3.5.1    3.5.2    3.5.1  test-outdated-output
+local-dev      0.0.3   linked   linked  test-outdated-output
+once           1.3.2    1.3.3    1.3.3  test-outdated-output
+```
+
+The output can be directly parsed and can be used to get the list of all
+outdated dependencies.
+
+### 2. FileExistBear
+
+Issue: [coala/coala-bears#527][527]
+
+The bear will be responsible to check if the files exist in the project
+directory. The name of the files can be passed as a regex to the bear, to
+which the bear will check their existence.
+
+```py
+class FileExistBear(LocalBear):
+    def run(self, required_files: tuple,):
+        for file in required_files:
+            if not os.path.is_file():
+                message = 'File {} not found.'.format(file)
+                yield Result.from_values(origin=self, message=message,)
+```
+
+### 3. RegexLintBear
+
+Issue: [coala/coala-bears#1532][1532]
+
+The task of the bear is to check the regex in strings. This can be done using
+the AnnotationBear to detect all the regex strings and then check for the valid
+regex through the prepared algorithm for each type.
+
+For Python, the bear can be written as follows:
```

Comment:
please move the regex checking logic to a separate function in your example code, as it is reusable, and note that the algorithm for it is derived from https://github.com/myint/rstcheck/pull/26/files

Reply:
Also it should be noted that this will only work in Python language , and the bear needs to expose that it only supports Python language using `aspects`

Reply:
The correct place for a python specific algorithm would be in pyflakes, or in our bear PyFlakesBear (on gitlab).

As this is a Generic Bear project, you need to find a generic solution.